src/GafferOSL/ShadingEngine.cpp

```diff
@@ -299,10 +299,33 @@ class RenderState
 		{
 			m_pointIndex++;
 		}
+	
+		bool getMatrixToObject( OIIO::ustring name, Imath::M44f &result ) const
+		{
+			ShadingEngine::Transforms::const_iterator i = m_transforms.find( name );
+			if( i != m_transforms.end() )
+			{
+				result = i->second.toObjectSpace;
+				return true;
+			}
+			return false;
+		}
+
+		bool getMatrixFromObject( OIIO::ustring name, Imath::M44f &result ) const
+		{
+			ShadingEngine::Transforms::const_iterator i = m_transforms.find( name );
+			if( i != m_transforms.end() )
+			{
+				result = i->second.fromObjectSpace;
+				return true;
+			}
+			return false;
+		}
 
 	private :
 
 		size_t m_pointIndex;
+		const ShadingEngine::Transforms &m_transforms;
```

Comment:
The slower part of `flat_map` is insertion, but it does have a [fast constructor from presorted ranges]( http://www.boost.org/doc/libs/1_56_0/doc/html/boost/container/flat_map.html#idp29465184-bb), which is what we'll have from the `Transforms` coming via the constructor.